File: go/leetcode/00062.go
package main

func uniquePaths(m int, n int) int {
	board := make([][]int, m)
	for i := range board {
		board[i] = make([]int, n)
	}

	for i := range m {
		for j := range n {
			if i == 0 || j == 0 {
				board[i][j] = 0
			} else {
				board[i][j] = board[i-1][j] + board[i][j-1]
			}
		}
	}
	return board[m-1][n-1]
}

File: python/LeetCode/00062.py
def uniquePaths(self, m: int, n: int) -> int:
    # Initialize m * n board of zeros
    board = [[0] * n] * m

    for i in range(m):
        for j in range(n):
            if i == 0 or j == 0:
                board[i][j] = 1
            else:
                board[i][j] = board[i-1][j] + board[i][j-1]
    
    return board[m - 1][n - 1]
